Controller:master:db use sMacaddr instead of std::string

maksym-bielan opened the pull request:
The master database son::dbuses std::string types to represent MAC addresses. This leads to a lot of conversions to sMacAddr structures, and carries a risk that a string sneaks in somewhere that is not a valid MAC address.

So all MAC addresses currently are represented by sMacAddr structure

Signed-off-by: Maksym Bielan <maksym.bielan@globallogic.com>

arnout commented on common/beerocks/bcl/include/beerocks/bcl/network/network_utils.h:
```diff
@@ -37,6 +37,7 @@ class network_utils {
     static const std::string ZERO_IP_STRING;
     static const std::string ZERO_MAC_STRING;
     static const std::string WILD_MAC_STRING;
+    static const sMacAddr ZERO_MAC;
```
:+1:

arnout commented on controller/src/beerocks/master/db/db.h:
```diff
@@ -513,18 +508,6 @@ class db {
                     certification_tx_buffer.get());
     };
 
-    struct bss_info_conf_t {
```
I think this got removed accidentally while rebasing...

arnout reviewed (commented):
Although it looks OK too far, I think it's way too ambitious to try to convert everything in a single commit.

Instead, handle just a few functions and their callers at a time.

maksym-bielan committed:
Controller:master:db use sMacaddr instead of std::string

The master database son::db uses std::string types to represent MAC
addresses. This leads to a lot of conversions to sMacAddr structures,
and carries a risk that a string sneaks in somewhere that is not a
valid MAC address.

Changed std::string to sMacAddr for son::db::has_node function and
overload get_node function to work with sMacAddr also.

Signed-off-by: Maksym Bielan <maksym.bielan@globallogic.com>

arnout commented on controller/src/beerocks/master/son_master_thread.cpp:
```diff
@@ -1106,7 +1106,7 @@ bool master_thread::handle_cmdu_1905_topology_discovery(Socket *sd,
     }
     auto al_mac = network_utils::mac_to_string(tlvAlMac->mac());
-    if (database.has_node(al_mac)) {
+    if (database.has_node(network_utils::mac_from_string(al_mac))) {
```
`al_mac` was just converted to string in the line above. Therefore, it would be better to use `tlvAlMac->mac()` directly here.

arnout reviewed (approved):
Looks very good already, I would like to merge it as is. You can still make a follow-up in a new PR.

The commit message should be wrapped at 72 columns - this is the case for the second paragraph but not the first.

The one comment I have is just a nitpick.
